### DISCUSSION TITLE
Fix for "FATAL Error: listen EACCES: permission denied 0.0.0.0:443" on wazuh-dashboard.service

### DISCUSSION THREAD

**Discussion by ThaBestStriker** [wazuh/wazuh#19560] [General]
After running an update, I've been trying to figure out for a fix on my wazuh-dashboard.service giving me a `FATAL Error: listen EACCES: permission denied 0.0.0.0:443` after running `systemctl status wazuh-dashboard`.

Many of the guides that I found with some google-fu provided a solution to run the service as root, which was not an acceptable solution for me. Instead, I did some research to figure out the source of the error and a solution for the problem.

First, I'm running Ubuntu 22.04.3 LTS with Wazuh 4.5.2 as a standalone server.
After running the systemctl command, I saw that some plugins were disabled and received the EACCES error. You can find the binary with: `sudo cat /etc/systemd/system/wazuh-dashboard.service | grep ExecStart`. 

The service executes the `/usr/share/wazuh-dashboard/bin/opensearch-dashboards` binary using `/etc/wazuh-dashboard/opensearch_dashboards.yml` to configure the service. I ran this binary as sudo to first confirm this was a permissions issue. In doing this, you receive the following error: `OpenSearch Dashboards should not be run as root.  Use --allow-root to continue.` Run again with `sudo /usr/share/wazuh-dashboard/bin/opensearch-dashboards --allow-root "-
c /etc/wazuh-dashboard/opensearch_dashboards.yml"`.

At this point, the server should start normally and can be accessed remotely via the URL. If it doesn't, you might have another issue. This is what many tutorials suggest as a "fix." To do this, you would edit the `/etc/systemd/system/wazuh-dashboard.service` ExecStart variable to include the `--allow-root` switch, listed above. But its never safe to run any service as root where possible, especially not one with an open connection to the internet. So I did more digging to find a solution that did not require running the service under root permissions.

I eventually ran into [this article](https://superuser.com/questions/710253/allow-non-root-process-to-bind-to-port-80-and-443), using noob's solution. I am not very experienced, as I'm using Wazuh as a simple home solution. While the other solutions listed might work great, I figured the safest method for me would be to limit the amount of system parameters that I change. That way, I'm less likely to make a change that affects another service.

To make the necessary change, I created an iptables rule to portfwd 443 to an ephemeral port:
`sudo iptables -t nat -I PREROUTING -d [serverIP] -p tcp --dport 443 -j REDIRECT --to-port [ephemeralPort]`

After this, I modified the `/etc/wazuh-dashboard/opensearch_dashboards.yml` file, changing `server.port: 443` to the ephemeral port number I used in the previous iptables rule.

Lastly, if you've created iptables filter rules, you'll need to add the ephemeral port to that ruleset and remove 443:
### Allows Inbound Traffic to Webserver on New and Established Connections ###
`sudo -I INPUT 2 -p tcp --dport [ephemeralPort] -d [serverIP] -m state --state NEW,ESTABLISHED,RELATED -j ACCEPT`

### Allows Outbound Traffic from Webserver on only Established and Related Connections ###
(You don't want NEW, because all outbound connections should already be established)
`sudo iptables -I OUTPUT 2 -p tcp --sport 15443 -s 143.42.232.90 -m state --state ESTABLISHED,RELATED -j ACCEPT`

(I use my Rule 1 to guarantee access to my local ssh session, so I don't accidentally lock myself out)

At this point, verify your current iptables configurations are correct:
`iptables -nvL --line-numbers`
Remove any previous rules that allowed traffic In/Out of Port 443, because your prerouting rule should handle all packets destined to 443 before the filter table takes a look at it:
`sudo iptables -D INPUT [RuleNumber]`
`sudo iptables -D OUTPUT [RuleNumber]`

At this point, everything should be properly configured to start the service:
`sudo systemctl start wazuh-dashboard.service`

Check the status of the service and make sure it starts properly, then check your dashboard from a remote computer to verify. If you experience any issues connecting, look at your iptables and verify your newly-implemented rules' packet counts are incrementing. If not, you might have a rule conflict in the filter table. If the NAT rule isn't incrementing, verify the syntax and make sure the nat kernel module is loaded:
`lsmod | egrep -i "nat|tables"`
If its loaded and correct, I unfortunately can't help.

Once you have verified the service starts properly and is accessible as intended, ensure you back up your current firewall config to a safe location:
`sudo iptables-save > [SafePath]/iptables.v4.bak`
Then ensure the rules have persistence, via whatever method you use.

And that's it! If you have any additional misinfo/concerns/suggestions regarding my configuration, let me know. I'm only making this because I never saw a guide or solution to resolve this issue, or possibly this issue is just so basic that its obvious to those who have a SysAdm background. This is my first guide for really anything computer-related, so let me know if you have any suggestions on improvement for future content.

**Comment by hron84** [comment]
I ran into this issue after a disk reorganization. Previously, wazuh-dashbaord worked flawless, but after migrating to a bigger disk, it stop running correctly and I don't know why. A freshly installed wazuh somehow manages to listen on 443 while running as wazuh-dashboard user.

**Update:** The solution was the following command:

```sh
 setcap 'cap_net_bind_service=+ep' /usr/share/wazuh-dashboard/node/bin/node
```

This command is included in the package's postinst commands and allows node to run on low ports without being root.

**Reply by ssit138543** [reply]
Our wazuh instance went down last night.  Wazuh dashboard service was not running for some reason.  The command you posted solved our issue.  Thank you.
